docs/docs/examples.md

```diff
@@ -0,0 +1,317 @@
+Hence, parsing and properly validating is a bit more elaborate:
+
+```kotlin linenums="1"
+Asn1Element.parse(customSequence.derEncoded).asStructure().let { root -> 
+
+  //↓↓↓ In reality, this would be a global constant; the same as in the previous snippet ↓↓↓
+  val rootTag = Asn1Element.Tag(1337uL, tagClass = TagClass.APPLICATION, constructed = true)
+  root.assertTag(rootTag) //throws on tag mismatch
+
+  val numberOfOps = root.nextChild().asPrimitive().decodeToUInt()
+  root.nextChild().asEncapsulatingOctetString().let { timestamp ->
+    val isRelative = timestamp.nextChild().asPrimitive()
+      .decode(TAG_TIME_RELATIVE) { Boolean.decodeFromAsn1ContentBytes(it) }
+
+    val time = if (isRelative) timestamp.nextChild().asPrimitive().decodeToUInt()
+    else timestamp.nextChild().asPrimitive().decodeToInstant()
+
+    if (timestamp.hasMoreChildren() || root.hasMoreChildren())
+      throw Asn1StructuralException("Superfluous Content")
+      
+    // Everything is parsed and validated
+    TODO("Create domain object from $numberOfOps, $isRelative, and $time")
+    }
+}
+```
+
+The above snippet performs the following validations:
+
+1. Line 5 asserts the tag of the root structure
+2. Line 7 ensures that the first child is an ASN.1 primitive tagged as INT, containing an unsigned integer
+3. Line 8 execution successfully guarantees that the second child is indeed an ASN.1 OCTET STRING encapsulating
+another ASN.1 structure.
+4. Lines 9-10 verify that the first child contained in the ASN.1 OCTET STRING
+    * is an ASN.1 primitive
+    * tagged with `TAG_TIME_RELATIVE`
+    * containing an ASN.1 boolean
+5. Line 12 ensures that the next child is an ASN.1 primitive, encoding an unsigned integer (in case an `UInt` is expected)
+6. Line 13 tackles the alternative and ensures that the next child contains a properly encoded ASN.1 time
+7. Lastly, lines 15-16 make sure no additional content is present, thus fully verifying the structure as a whole.
+
+
+## Issuing Binding Certificates
+
+!!! info  inline end
+    This example requires the _Supreme_ KMP crypto provider. Only _Signum_-specifics are illustrated using code snippets.
+
+We'll assume a JVM backend using _[WARDEN](https://github.com/a-sit-plus/warden)_ and trust anchors all set up correctly
+on the client apps.
+A common pattern in a mobile client setting in the context of banking or eID are so-called _binding certificates_ (or binding keys, but we'll stick to certificates here).
+Just assume a bank with a mobile client application: Customers are typically issued an activation token out-of-band
+(via mail, by the teller, …). This token is used to activate the app and transactions can then be authorized using biometrics.
+
+In settings as critical as eID and banking, the service operator typically wants to ensure that only uncompromised clients
+may access a service. To ensure this, the example described here relies on attestation.
```

Comment:
"on attestation" -- app, or key? "only uncompromised clients" sounds like app attestation, but i think what we're doing is key attestation?

Reply:
though i guess that's a distinction made at the back-end, since "app attestation" is just attribute checking in the key attestation certificate?